src/edit/items/BoundWithFieldRow.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Loading } from '@folio/stripes/components';
+
+import FieldRow from '../../components/RepeatableField/FieldRow';
+import useHoldingsQueryByHrids from '../../hooks/useHoldingsQueryByHrids';
+import useInstancesQuery from '../../hooks/useInstancesQuery';
+
+// Fetches holdings and instances info given in the input holdings HRID
+const BoundWithFieldRow = ({ fields, ...rest }) => {
+  const data = fields.value ? fields.value : [];
+
+  // Load the holdings records matching the input HRIDs
+  const holdingsRecordHrids = data.map(boundWithTitle => boundWithTitle.briefHoldingsRecord.hrid);
+  const { isLoading: isHoldingsLoading, holdingsRecords } = useHoldingsQueryByHrids(holdingsRecordHrids);
+
+  // Load the instance records matching the holdings' instanceIds.
+  const instanceIds = holdingsRecords.map(record => record.instanceId);
+  const instances = useInstancesQuery(instanceIds);
+  if (isHoldingsLoading || !instances.isSuccess) return <Loading size="large" />;
+
+  // Enrich the data render in the FieldRow
+  data.map(boundWithTitle => {
```

Comment:
@maccabeelevine this is nitpicking but should we replace `map` with `forEach` since we are just looping here mutating `boundWithTitle` without returning any value at the end.